### PR TITLE
Token balances contract

### PR DESCRIPTION
**Pull request by karteufel**
Independent. Adds a contract with 'get token balances'-related methods

### CODE REVIEW

File: contracts/TokenStoreBalances.sol [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    revert();
+  }
+
+  // Limit withdrawals to the contract creator
+  modifier isAdmin() {
+    require(msg.sender == admin);
+    _;
+  }
+
+  // Backup withdraw, in case ETH gets in here
```
Comment (tokenstoredev) [inline contracts/TokenStoreBalances.sol]: Let's remove it, payable is disabled.

File: contracts/TokenStoreBalances.sol [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  function withdraw() external isAdmin {
+    admin.transfer(address(this).balance);
+  }
+
+  // Backup withdraw, in case ERC20 tokens get in here
+  function withdrawToken(address token, uint amount) external isAdmin {
```
Comment (tokenstoredev) [inline contracts/TokenStoreBalances.sol]: Let's remove it, not sure why anybody would send tokens to this address. We can remove admin part and constructor completely after that.

File: contracts/TokenStoreBalances.sol [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+   // is it a contract and does it implement balanceOf
+    if(tokenCode > 0 && token.call(bytes4(0x70a08231), user)) {    // bytes4(keccak256("balanceOf(address)")) == bytes4(0x70a08231)
+      return Token(token).balanceOf(user);
+    } else {
+      return 0; // not a valid token, return 0 instead of error
+    }
+  }
+
+ /* Combine walletBalances() and deltaBalances() to get both exchange and wallet balances for multiple tokens.
```
Comment (tokenstoredev) [inline contracts/TokenStoreBalances.sol]: Nit: remove?

File: contracts/TokenStoreBalances.sol (hunk 1 above)
Comment (karteufel) [reply contracts/TokenStoreBalances.sol]: Done

File: contracts/TokenStoreBalances.sol (hunk 2 above)
Comment (karteufel) [reply contracts/TokenStoreBalances.sol]: Done

File: contracts/TokenStoreBalances.sol (hunk 3 above)
Comment (karteufel) [reply contracts/TokenStoreBalances.sol]: Done

**Comment by karteufel** [comment]
Added changes and rechecked locally

**Pull request merged**